### PR TITLE
fix: Misc fixes handling correct image name and identifier

### PR DESCRIPTION
**Pull request by nderjung**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

This PR includes several fixes surrounding issues with unspecified or misspecified image names either lacking entire name or an identifier.

### CODE REVIEW

File: internal/cli/kraft/cloud/compose/build/build.go [hunk 1]
```diff
@@ -138,6 +138,9 @@ func Build(ctx context.Context, opts *BuildOptions, args ...string) error {
 				pkgName,
 			)
 		}
+		if !strings.HasSuffix(pkgName, ":latest") {
+			pkgName = strings.TrimSuffix(pkgName, ":latest")
+		}
```
Comment (craciunoiuc) [inline internal/cli/kraft/cloud/compose/build/build.go]: so you remove it if it's not set?

Comment (nderjung) [reply internal/cli/kraft/cloud/compose/build/build.go]: Whoops

Comment (nderjung) [reply internal/cli/kraft/cloud/compose/build/build.go]: I've now dropped this commit. The subsequent commit fixes this issue implicitly.

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**